types/connector.d.ts

```diff
@@ -3,28 +3,384 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-import {Callback, DataSource, Options, PromiseOrVoid} from '..';
+import { AnyObject } from 'strong-globalize/lib/config';
+import {Callback, DataSource, Filter, ModelBase, ModelBaseClass, ModelDefinition, ModelProperties, Options, PromiseOrVoid, PropertyDefinition, PropertyType, Schema, Where} from '..';
 
-// Copyright IBM Corp. 2018. All Rights Reserved.
-// Node module: loopback-datasource-juggler
-// This file is licensed under the MIT License.
-// License text available at https://opensource.org/licenses/MIT
+export interface ConnectorSettings extends Options {
+  name?: string;
+  /**
+   * Overrides {@link ConnectorSettings.adapter} if defined.
+   */
+  connector?: ConnectorStatic | string;
+  /**
+   * @deprecated Use {@link ConnectorSettings.connector} instead.
+   */
+  adapter?: ConnectorStatic | string;
+  database: string;
+  connectionTimeout?: number;
+  maxOfflineRequests?: number;
+  lazyConnect?: boolean;
+  debug?: boolean;
+
+  // Postgres-specific
+  defaultIdSort?: boolean | 'numericIdOnly';
+  onError?: (err: Error | unknown) => unknown | 'ignore';
+  // END Postgres-specific
+}
```

Comment:
Add `loopback-connector-mssql`-specific settings:

```suggestion
  tableNameID?: string
}
```

#TODO: Figure out what this setting does.